### DISCUSSION TITLE
Uncaught (in promise) Error: Connection got disposed.

### DISCUSSION THREAD

**Discussion by habibali95** [eclipse-theia/theia#8545] [General]
Im using k8s ingress to run theia, ive tried many version and i always get these errors, is there any reasons why could this happen? 

root ERROR Uncaught Exception:  Error: Connection got disposed.
root ERROR Error: Connection got disposed.
    at Object.dispose (/home/theia/node_modules/vscode-jsonrpc/lib/main.js:904:25)
    at /home/theia/node_modules/vscode-ws-jsonrpc/lib/socket/connection.js:14:41
    at CallbackList.invoke (/home/theia/node_modules/vscode-jsonrpc/lib/events.js:62:39)
    at Emitter.fire (/home/theia/node_modules/vscode-jsonrpc/lib/events.js:121:36)
    at closeHandler (/home/theia/node_modules/vscode-jsonrpc/lib/main.js:240:26)
    at CallbackList.invoke (/home/theia/node_modules/vscode-jsonrpc/lib/events.js:62:39)
    at Emitter.fire (/home/theia/node_modules/vscode-jsonrpc/lib/events.js:121:36)
    at WebSocketMessageReader.fireClose (/home/theia/node_modules/vscode-jsonrpc/lib/messageReader.js:111:27)
    at WebSocketMessageReader.fireClose (/home/theia/node_modules/vscode-ws-jsonrpc/lib/socket/reader.js:67:19)
    at /home/theia/node_modules/vscode-ws-jsonrpc/lib/socket/reader.js:24:18

**Comment by habibali95** [comment]
and the status goes offline for some time..

**Comment by habibali95** [comment]
and then this happens randomly, 
root INFO [nsfw-watcher: 560] Started watching: /root/.theia
root INFO [5d75ac88-ce19-47b9-9aee-eefaa82795ab] Sync of 0 plugins took: 98.8 ms
root INFO [5d75ac88-ce19-47b9-9aee-eefaa82795ab] Load contributions of 0 plugins took: 0.1 ms
root INFO [hosted-plugin: 573] PLUGIN_HOST(573) starting instance
root INFO [nsfw-watcher: 460] Started watching: /home/project
root INFO [nsfw-watcher: 460] Started watching: /home/project
root ERROR [nsfw-watcher: 460] Failed to watch "/home/project": Inotify limit reached
root ERROR [nsfw-watcher: 460] Failed to watch "/home/project": Inotify limit reached
root INFO [nsfw-watcher: 460] Stopped watching: /home/project
root INFO [nsfw-watcher: 460] Stopped watching: /home/project
root INFO [nsfw-watcher: 560] Started watching: /home/project
root INFO [nsfw-watcher: 560] Started watching: /home/project
root ERROR [nsfw-watcher: 560] Failed to watch "/home/project": Inotify limit reached
root ERROR [nsfw-watcher: 560] Failed to watch "/home/project": Inotify limit reached
root INFO [nsfw-watcher: 560] Stopped watching: /home/project
root INFO [nsfw-watcher: 560] Stopped watching: /home/project
root INFO Changed application state from 'ready' to 'closing_window'.
root INFO >>> Storing the layout...
root INFO <<< The layout has been successfully stored.
root INFO >>> Stopping frontend contributions...
root INFO <<< All frontend contributions have been stopped.

**Reply by monaka** [reply]
I suppose the value of `inotify` on your Linux kernel is too small.
Please google with `inotify sysctl` to get more details.

**Comment by amiramw** [comment]
We had issues with nginx ingress where, if I recall, each routing change caused disconnetion.

**Comment by vince-fugnitto** [comment]
@habibali95 I believe you may want to increase your `inotify` watchers for your host: https://github.com/eclipse-theia/theia/issues/5842#issuecomment-517927455. By default the number is quite low (I updated it for myself when working on [Linux](https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#linux)).

**Reply by habibali95** [reply]
thanks @vince-fugnitto , i tried increasing inotify and it didn't fix the issue..

**Comment by RomanNikitenko** [comment]
Maybe related to https://github.com/TypeFox/vscode-ws-jsonrpc/issues/24

**Comment by habibali95** [comment]
hello @amiramw, is there any fix or workaround for this issue?

**Reply by amiramw** [reply]
This thread is relevant: https://github.com/kubernetes/ingress-nginx/issues/2647#issuecomment-397468148
I see it is closed now so maybe it is solved.
We don't use ingress nginx anymore.

**Reply by monaka** [reply]
I'm using Traefik 1.7 and it also similar issue in a particular situation.
Traefik 1.7 can't keep connection if broken Ingress resources.
You can check broken Ingresses by watching logs.
`{"level":"warning","msg":"Endpoints not available for` means broken.
